src/controllers/authController.js

```diff
@@ -65,8 +67,43 @@ const logout = async (req, res) => {
   res.status(StatusCodes.OK).json({ msg: 'user logged out!' });
 };
 
+//forgotPassword endpoint
+const forgotPassword = async (req, res) => {
+  const { email } = req.body;
+
+  if (!email) {
+    throw new CustomError.BadRequestError('Please provide an email');
+  }
+
+  const user = await User.findOne({ email });
+
+  if (!user) {
+    throw new CustomError.NotFoundError('User not found');
+  }
+
+  // Generate a random token
+  const resetToken = crypto.randomBytes(20).toString('hex');
+  const resetExpires = Date.now() + 15 * 60 * 1000; // 15 minutes from now
+
+  // Update user's reset token and expiration date
+  user.passwordResetToken = resetToken;
+  user.passwordResetExpiresOn = new Date(resetExpires);
+  await user.save();
+
+  // Send reset password email
+  const emailTemplate = 'forgotPassword'; // to adjust this based on our email template name
+  const recipient = user.email;
+  const localVariables = {
+    resetToken,
+  };
+  await emailSender(emailTemplate, recipient, localVariables);
```

Comment:
`emailSender` exports an object now (`module.exports = { sendTestEmail, sendForgotPasswordEmail };`) so your require here is an object, we need to tell it which function to call otherwise we'll get the following error when we call the endpoint:

![](https://github.com/Code-the-Dream-School/dd-prac-team4-back/assets/9094098/3e1686fd-9163-40be-903e-119a4f436f7e)

change required:

```suggestion
  await emailSender.sendForgotPasswordEmail(emailTemplate, recipient, localVariables);
```